File: CS/TestMyButtonEdit/RepositoryItemMyButtonEdit.cs
using System;
using System.ComponentModel;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Controls;





namespace DevExpress.MyControl
{
    [UserRepositoryItem("RegisterMyButtonEdit")]
    public class RepositoryItemMyButtonEdit : RepositoryItemButtonEdit
    {
        internal const string EDITORTypeName = "MyButtonEdit";
        public override string EditorTypeName
        {
            get
            {
                return EDITORTypeName;
            }
        }

        [Browsable(false)]
        public new MyButtonEdit OwnerEdit
        {
            get
            {
                return base.OwnerEdit as MyButtonEdit;
            }
        }


        static RepositoryItemMyButtonEdit()
        {
            RegisterMyButtonEdit();
        }



        public RepositoryItemMyButtonEdit()
            : base()
        {
        }



        public override void CreateDefaultButton()
        {
            base.CreateDefaultButton();
            EditorButton eb = new EditorButton(ButtonPredefines.Glyph);
            eb.IsDefaultButton = true;
            Buttons.Add(eb);
        }



        public static void RegisterMyButtonEdit()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EDITORTypeName,
            typeof(MyButtonEdit), typeof(RepositoryItemMyButtonEdit),
            typeof(MyButtonEditViewInfo), new MyButtonEditPainter(), true, null));
        }
    }
}

File: VB/TestMyButtonEdit/RepositoryItemMyButtonEdit.vb
Imports System
Imports System.ComponentModel
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Controls


Namespace DevExpress.MyControl
    <UserRepositoryItem("RegisterMyButtonEdit")> _
    Public Class RepositoryItemMyButtonEdit
        Inherits RepositoryItemButtonEdit

        Public Const EDITORTypeName_Renamed As String = "MyButtonEdit"
        Public Overrides ReadOnly Property EditorTypeName() As String
            Get
                Return EDITORTypeName_Renamed
            End Get
        End Property

        <Browsable(False)> _
        Public Shadows ReadOnly Property OwnerEdit() As MyButtonEdit
            Get
                Return TryCast(MyBase.OwnerEdit, MyButtonEdit)
            End Get
        End Property


        Shared Sub New()
            RegisterMyButtonEdit()
        End Sub

        Public Sub New()
            MyBase.New()
        End Sub

        Public Overrides Sub CreateDefaultButton()
            MyBase.CreateDefaultButton()
            Dim eb As New EditorButton(ButtonPredefines.Glyph)
            eb.IsDefaultButton = True
            Buttons.Add(eb)
        End Sub

        Public Shared Sub RegisterMyButtonEdit()
            Dim img As Image = Nothing
            EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EDITORTypeName_Renamed, GetType(MyButtonEdit), GetType(RepositoryItemMyButtonEdit), GetType(MyButtonEditViewInfo), New MyButtonEditPainter(), True, img))
        End Sub
    End Class
End Namespace
